src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+transitional, mostly procedural system on its way to being summarised as configuration - as numerous Infusion features have
+been in their turn over the years.
+
+## Notes on interoperability
+
+As commented [below](#commentary-on-requirements), Infusion promises have taken a different set of tradeoffs to many of those elsewhere in the industry. In terms of interoperability,
+Infusion promises are at least universally recognised as a "[foreign thenable](https://github.com/cujojs/when/blob/master/docs/api.md#when)" and hence can be easily adapted into promises of any of the other libraries.
+In terms of promise algorithms, since Infusion promises meet a weaker contract than usual, Infusion promises cannot safely be supplied to the promise
+algorithms of other libraries without adaptation. However, promises from foreign libraries can easily be used within Infusion's algorithms.
+
+## Core Promises API
+
+### fluid.promise()
+
+* Returns: `{Promise}`
+
+Construct a fresh promise. This is the only point at which fresh promises are constructed within the core API.
+
+### promise.then(onResolve, onReject)
+
+Adds handlers to either or both of the `resolve` or `reject` actions of the promise. Note that if the promise has already been
+rejected or resolved, the appropriate handler will be notified immediately on this registration.
+
+* `onResolve: {Function ({Any}) →  None}` A callback to receive the successfully resolved value of the promise
+* `onReject: {Function ({Error}) →  None}` A callback to receive a rejection of the promise, in the case its resolution fails.
+
+### promise.resolve(value)
+
+Resolves the promise successfully, yielding the value `value` to any listeners which were previously registered as the first
+argument of `promise.then`. Any further attempts to call either `resolve` or `reject` will signal an assertion failure.
+
+* `value: {Any}` The value to be supplied as the resolution value of the promise
+
+### promise.reject(error)
+
+Rejects the promise, yielding the error `error` to any listeners which were previously registered as the second 
+argument of `promise.then`. Any further attempts to call either `resolve` or `reject` will signal an assertion failure.
+
+* `error: {Object|Error}` The value to be supplied as the rejection reason for the promise. It is recommended that this be an
+object with a boolean entry `isError: true` and a String `message` summarising the reason for the rejection. 
+
+### promise.disposition
+
+The current disposition of the promise may be inspected at any time. This is a `String` value which encodes which, if any, of
+`resolve` or `reject` have been received by the promise. At the fresh construction of the promise, the member `disposition` holds
+the value `undefined`. If the promise has received `resolve`, `disposition` will hold the string `"resolve"`, or if the
+promise has received `reject`, `disposition` will hold the string `"reject"`.
+
+### Commentary on core API
```

Comment:
This information doesn't actually strike me as a commentary on the core api, but rather just a description of promises in Infusion. As such, it would be better suited to the beginning of the file, instead of after the api details. As a reader, I would have gotten more out of the API description above if I already had *this* information in my head, as context.